src/cpp/include/openvino/genai/whisper_generation_config.hpp

```diff
@@ -15,28 +16,14 @@ namespace genai {
 /**
  * @brief Structure to keep whisper generation config parameters.
  */
-class OPENVINO_GENAI_EXPORTS WhisperGenerationConfig {
+class OPENVINO_GENAI_EXPORTS WhisperGenerationConfig : public GenerationConfig {
 public:
     WhisperGenerationConfig() = default;
     explicit WhisperGenerationConfig(const std::filesystem::path& json_path);
 
-    // Generic
-
-    // the maximum length the generated tokens can have. Corresponds to the length of the input prompt +
-    // `max_new_tokens`. Its effect is overridden by `max_new_tokens`, if also set.
-    size_t max_new_tokens = SIZE_MAX;
-    // the maximum numbers of tokens to generate, excluding the number of tokens in the prompt.
-    // max_new_tokens has priority over max_length.
-    size_t max_length = SIZE_MAX;
-
-    // Whisper specific
-
     // Corresponds to the ”<|startoftranscript|>” token.
     int64_t decoder_start_token_id = 50258;
 
-    // End of stream token id.
-    int64_t eos_token_id = 50257;
```

Comment:
I see that within Whisper code we use this value in some places like:
```c++
            // cannot be normal text token
            for (size_t i = 0; i < config.eos_token_id; i++) {
                logits_data[i] = -std::numeric_limits<float>::infinity();
            }
```
do we need to handle multiple stop tokens, then?

Reply:
The `eos_token_id` used here not as a stop criteria but as model characteristic. It's used to force double timestamps tokens when return timestamps enabled. HF doesn't handle multiple eos tokens in timestamps logits processor as well.